hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaControllerManagement.java

```diff
@@ -657,6 +657,16 @@ private void handleFinishedAndStoreInTargetStatus(final JpaAction action) {
             ConcurrencyFailureException.class }, maxAttempts = Constants.TX_RT_MAX, backoff = @Backoff(delay = Constants.TX_RT_DELAY))
     public Target updateControllerAttributes(final String controllerId, final Map<String, String> data,
             final UpdateMode mode) {
+
+        /*
+            Constraint is not validated by EclipseLink. Therefore, constraints have to be checked here.
+         */
+        if (data.entrySet().stream()
+                .anyMatch(e -> !JpaControllerManagement.isAttributeEntryValid(e))) {
```

Comment:
Use method reference?

Reply:
I would have to change the logic of `isAttributeEntryValid` to `is AttributeEntryInvalid` to use method reference. Therefore, I'd like to keep it that way.